group-coordinator/src/test/java/org/apache/kafka/coordinator/group/modern/consumer/TopicRegexResolverTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.modern.consumer;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.internals.Plugin;
+import org.apache.kafka.common.metadata.TopicRecord;
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.coordinator.common.runtime.CoordinatorMetadataImage;
+import org.apache.kafka.coordinator.common.runtime.KRaftCoordinatorMetadataImage;
+import org.apache.kafka.image.MetadataDelta;
+import org.apache.kafka.image.MetadataImage;
+import org.apache.kafka.image.MetadataProvenance;
+import org.apache.kafka.server.authorizer.Action;
+import org.apache.kafka.server.authorizer.AuthorizationResult;
+import org.apache.kafka.server.authorizer.Authorizer;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TopicRegexResolverTest {
+
+    private final Logger log = LoggerFactory.getLogger(TopicRegexResolverTest.class);
+
+    private static CoordinatorMetadataImage imageWithTopics(String... topicNames) {
+        MetadataDelta delta = new MetadataDelta(MetadataImage.EMPTY);
+        for (String topic : topicNames) {
+            delta.replay(new TopicRecord()
+                .setName(topic)
+                .setTopicId(Uuid.randomUuid()));
+        }
+        MetadataImage image = delta.apply(MetadataProvenance.EMPTY);
+        return new KRaftCoordinatorMetadataImage(image);
+    }
+
+    @Test
+    public void testBasicMatching() {
+        CoordinatorMetadataImage image = imageWithTopics("foo", "bar", "baz", "qux");
```

Comment:
nit: Could we reuse the `MetadataImageBuilder` that we use in other group coordinator tests?